ngraph/frontend/onnx_import/include/onnx_import/editor/detail/subgraph_extraction.hpp

```diff
@@ -0,0 +1,152 @@
+//*****************************************************************************
+// Copyright 2017-2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <map>
+#include <set>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+namespace ONNX_NAMESPACE
+{
+    class GraphProto;
+    class NodeProto;
+    class ValueInfoProto;
+} // namespace ONNX_NAMESPACE
+
+namespace ngraph
+{
+    namespace onnx_import
+    {
+        /// \brief Defines an edge connected to an input of any node in the graph.
+        ///        It consists of a node index in the processed ONNX model and the input name.
```

Comment:
Maybe just to clarify is worth to emphasize that nodes numeration start from 0 (from inputs to outputs in topological order) - if I understand correctly